PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/functions/component/table/AppSettingsTableUtils.java

```diff
@@ -0,0 +1,182 @@
+            @Override
+            public void valueChanged(ListSelectionEvent listSelectionEvent) {
+                final String prompt = AzureFunctionsConstants.getAppSettingHint(appSettingsTable.getSelectedKey());
+                promptPanel.setText(prompt);
+            }
+        });
+
+        appSettingsTable.addFocusListener(new FocusListener() {
+            @Override
+            public void focusGained(FocusEvent focusEvent) {
+                final String prompt = AzureFunctionsConstants.getAppSettingHint(appSettingsTable.getSelectedKey());
+                promptPanel.setText(prompt);
+            }
+
+            @Override
+            public void focusLost(FocusEvent focusEvent) {
+                promptPanel.setText("");
+            }
+        });
+
+        final ToolbarDecorator tableToolbarDecorator = ToolbarDecorator.createDecorator(appSettingsTable)
+                .addExtraActions(btnAdd, btnRemove, importButton, exportButton).setToolbarPosition(ActionToolbarPosition.RIGHT);
+        final JPanel tablePanel = tableToolbarDecorator.createPanel();
+        final GridConstraints tableConstraint = new GridConstraints(0, 0, 1, 1, 0, GridConstraints.FILL_BOTH, 7, 7, null, null, null);
+        result.add(tablePanel, tableConstraint);
+        result.setMinimumSize(new Dimension(-1, 100));
+        return result;
+    }
+
+    public static Map<String, String> getAppSettingsFromLocalSettingsJson(File target) {
+        final Map<String, String> result = new HashMap<>();
+        try {
+            final JsonObject jsonObject = readLocalSettingsJsonFile(target);
+            final JsonObject valueObject = jsonObject.getAsJsonObject(LOCAL_SETTINGS_VALUES);
+            valueObject.entrySet().forEach(entry -> result.put(entry.getKey(), entry.getValue().getAsString()));
+            return result;
+        } catch (JsonParseException e) {
+            return new HashMap<>();
+        }
+    }
+
+    public static void exportLocalSettingsJsonFile(File target, Map<String, String> appSettings) throws IOException {
+        if (target == null) {
+            return;
+        }
+        if (!target.exists()) {
+            target.createNewFile();
+        }
+        final Gson gson = new GsonBuilder().setPrettyPrinting().disableHtmlEscaping().create();
+        JsonObject jsonObject = readLocalSettingsJsonFile(target);
+        if (jsonObject == null) {
+            jsonObject = gson.fromJson(DEFAULT_LOCAL_SETTINGS_JSON, JsonObject.class);
+        }
+        final JsonObject valueObject = new JsonObject();
+        appSettings.entrySet().forEach(entry -> valueObject.addProperty(entry.getKey(), entry.getValue()));
+        jsonObject.add(LOCAL_SETTINGS_VALUES, valueObject);
+        try (Writer writer = new FileWriter(target)) {
+            gson.toJson(jsonObject, writer);
+        }
+    }
+
+    private static JsonObject readLocalSettingsJsonFile(File target) {
+        try (FileInputStream fis = new FileInputStream(target);
```

Comment:
add a todo mark: we need to use `azure-maven-plugin-common/src/main/java/com/microsoft/azure/maven/common/utils/JsonUtils.java` when it is moved to azure tools common

Reply:
Done